docstore/memdocstore/mem.go

```diff
@@ -68,6 +68,12 @@ type Options struct {
 	// When the collection is closed, its contents are saved to the file.
 	Filename string
 
+	// AllowNestedSliceQueries allows querying with nested slices.
+	// This makes the memdocstore more compatible with MongoDB,
+	// but other providers may not support this feature.
+	// See https://github.com/google/go-cloud/pull/3511 for more details.
```

Comment:
Don't refer to a PR for documentation. Make the doc self-contained by explaining how this changes the behavior, that is, precisely what "querying with nested slices" means.

You can refer to the PR in an implementation comment so developers can understand the context.